bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/api/HeosSystem.java

```diff
@@ -502,45 +269,47 @@ public void setConnectionPort(int connectionPort) {
      *
      * @author Johannes Einig
      */
-    public class KeepAliveRunnable implements Runnable {
+    private class KeepAliveRunnable implements Runnable {
+
         @Override
         public void run() {
             try {
-                if (sendCommand.isConnectionAlive()) {
-                    logger.debug("Sending Heos Heart Beat");
-                    if (!sendCommand.send(command().heartbeat())) {
-                        logger.debug("Connection to HEOS Network lost!");
-                        restartConnection();
+                if (sendCommand.isHostReachable()) {
+                    long timeSinceLastEvent = System.currentTimeMillis() - eventController.getLastEventTime();
+                    logger.debug("Time since latest event: {} s", timeSinceLastEvent / 1000);
+
+                    logger.debug("Sending HEOS Heart Beat");
+                    HeosResponseObject<Void> response = send(HeosCommands.heartbeat());
+                    if (timeSinceLastEvent < LAST_EVENT_THRESHOLD && response.result) {
```

Comment:
Is the `LAST_EVENT_THRESHOLD' a debugging feature? Or what is it good for?

Reply:
Good question. I had multiple times healthy-looking connections with the HEOS which stopped transmitting events. However, the send-connection was still fine so no reconnect was ever triggered. So I ended up with this to resolve it. I would propose to keep it there, okay?